### ISSUE TITLE
Cucumber tests should log to file

### ISSUE THREAD

**Issue by hferentschik** [minishift/minishift#607]
To make it easier to debug Cucumber test failures, we should add a logging framework which logs to file (in order to not interfere with the default output of GoDog). In this file we should then for example log the commands executed and their results, etc.

**Comment by agajdosi** [comment]
Took a brief look onto this today. I am successful with using StdoutPipe - `myout, _ := cmd.StdoutPipe()` for stdout of command. This gives a possibility to:

+ log the stdout into file realtime, having each line with exact time
+ could be also handy to implement an option of live preview of executed commands

Current code is quite "hacky", will clean it tomorrow and probably give some new info or WIP PR.

**Comment by agajdosi** [comment]
Hi @hferentschik, if you have time, could you please take a look on [these lines](https://github.com/agajdosi/minishift/blob/issue607/test/integration/util/util.go#L46-L96), if my way is completely off or is fine?

I took a look on Tee struct and its methods which you mentioned on IRC, but I got no idea where/how it should be used? If better way is to implement Tee, please let me know, I will try to look better or will need to call for bjn call.

**Comment by gbraad** [comment]
this makes the runCommand function much harder to understand and reason
about.
Is it not possible to wrap the actual call to runCommand and just use the
stdOut, stdErr, etc afrom return and store these?

**Comment by agajdosi** [comment]
So I got logging into file working. One last thing is: `panic: interface conversion: interface {} is *gherkin.ScenarioOutline, not *gherkin.Scenario`, which I will fix tomorrow, forgot that these are two different types.

This is how log currently looks:

```
16:10:27.610979    info: Log initiated
16:10:27.611804    info: ----- scenario Version -----
16:10:27.611920 command: /home/agajdosi/golang/bin/minishift version
16:10:27.674226  stdout: -- Installing default add-ons ... OK
minishift v1.4.1+e5cc0e8
16:10:27.674660    info: ----- scenario Config -----
16:10:27.674719 command: /home/agajdosi/golang/bin/minishift config view
16:10:27.688485  stdout: 
16:10:27.689006    info: ----- scenario Fail -----
16:10:27.689077 command: /home/agajdosi/golang/bin/minishift configaro
16:10:27.699849  stdout: 
16:10:27.699869  stderr: Error: unknown command "configaro" for "minishift"
Run 'minishift --help' for usage.
unknown command "configaro" for "minishift"
16:10:27.700063 command: /home/agajdosi/golang/bin/minishift delete
16:10:27.712839  stdout: Running this command requires an existing 'minishift' VM, but no VM is defined.
16:10:27.712922 command: /home/agajdosi/golang/bin/minishift status
16:10:27.724952  stdout: Does Not Exist
```

There could be 2 improvements:
- Instead of using `util.IntegrationLog.Print("things-to-log")` I could create a specialized function which would take care of multi-line things to be logged - this could format all other new lines with x whitespaces so it would be aligned.
- Implement also the pipe solution which would give us timestamps on every line of stdout and stderr - for example in start we would have detailed time information where it took how much time, not just that whole start took 3 minutes.

**Comment by hferentschik** [comment]
Looks good to me. I definitely like the output you posted. Let's get this in and then we can improve. Some helper method for formatting sounds reasonable (for example for multi line log statement, if this is an issue for some reason). Still skeptical towards the pipe solution, at least in the form we looked at it. The code looked quite complicated. If we go down this route, I think we can do it in a more readable fashion. I definitely understand so what you want to achieve. It would give some timings for the various steps being reported to stdout. I still think so that in this case you are better off with a slightly modified version of Tee. ATM, Tee is geared towards os.Stdout and os.Stderr, but one can make this a bit more generic and ass in output and error handle, in which case you can split arbitrary stream with a Tee.

**Comment by agajdosi** [comment]
@hferentschik Great to hear. Ok, so I will add formatting helper, could be PutInLog(string), which will help with multiline, it will improve readability of log file and that is good each time. But for one line messages Print() will be still usable. Also will try to get there some `----- Cleaning up -----` to draw a border between last scenario and cleaning commands.

With the pipeline I agree with you, lets get this in and then I can take the pipelines as low priority "hobby" task. Also we will get some ideas on improvements during the time, that is sure.

Thanks for feedback.

**Comment by hferentschik** [comment]
Merged via pull request #1306
